### PR TITLE
feat(atomic): cRGA refactor + atomic-insight-generated-answer component 

### PR DESCRIPTION
**Pull request by nathanlb**
Moved most of the code for the atomic-generated-answer component into the common directory and refactored the core functionality into a "common" class.

The resulting use case-specific components are quite small and tidy.

The `atomic-insight-generated-answer` component is new.

### CODE REVIEW

**Review by louis-bompart — APPROVED** [review]
lgtm, a lil nit pic

File: packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    return this.props.getBindings().i18n.t(key);
+  }
+
+  private get loadingClasses() {
+    return 'my-3';
+  }
+
+  private get contentClasses() {
+    return 'mt-0 mb-4 border border-neutral shadow-lg p-6 bg-background rounded-lg p-6 text-on-background';
+  }
```
[the remark this answers is not in the document]
Comment (nathanlb) [reply packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx:139]: Yes good point

File: packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  copyError = false;
+
+  /**
+   * The answer style to apply when the component first loads.
+   * Options:
+   *   - `default`: Generates the answer without additional formatting instructions.
+   *   - `bullet`: Requests that the answer is formatted as a bulleted list.
+   *   - `step`: Requests that the answer is formatted as a series of step-by-step instructions.
+   *   - `concise`: Requests that the generated answer is as concise as possible.
+   */
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx:77]:
```suggestion
  /**
   * The answer style to apply when the component first loads.
   * Options:
   *   - `default`: Generate the answer without specific formatting instructions.
   *   - `bullet`: Generate the answer as a bulleted list.
   *   - `step`: Generate the answer as step-by-step instructions.
   *   - `concise`: Generate the answer as briefly as possible.
   */
```

File: packages/atomic/src/components/insight/atomic-insight-generated-answer/atomic-insight-generated-answer.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  @Element() private host!: HTMLElement;
+
+  @State()
+  copied = false;
+
+  @State()
+  copyError = false;
+
+  /**
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/insight/atomic-insight-generated-answer/atomic-insight-generated-answer.tsx:73]: See my comment below

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx (hunk 2 above)
Comment (nathanlb) [reply packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx:77]: much better

**Review by drosenfeld-coveo — APPROVED** [review]

**Pull request merged**